### PR TITLE
Service documentation

### PR DESCRIPTION
**Pull request by antho1404**
Dependency: https://github.com/liteflow-labs/liteflow-js/pull/131

Add new documentation related to the services

### CODE REVIEW

File: docs/pages/service/index.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+title: 'Introduction'
+---
+
+# Liteflow services
+
+Liteflow enables customization of application behavior through the use of services, also known as micro-services. By utilizing directives from the application, Liteflow's infrastructure becomes highly flexible and powerful.
+
+## Services in Liteflow
+
+Liteflow communicates with the application through requests as HTTPS JSON payloads. The application can then respond with configurations for specific actions.
```
Comment (NicolasMahe) [inline docs/pages/service/index.mdx]:
```suggestion
Liteflow communicates with the application through HTTPS requests with JSON payloads. The application responds with configurations for the specific actions.
```

File: docs/pages/service/index.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Configuring a Service
+
+To start using a service in Liteflow, follow these simple steps:
+
+1. Identify the desired service to respond to.
+2. Create a public HTTPS URL as the endpoint for the service.
+3. Parse incoming payloads and respond with a 200 status code for each request received from Liteflow.
```
Comment (NicolasMahe) [inline docs/pages/service/index.mdx]:
```suggestion
3. Parse incoming payloads and respond with the expected response with a 200 status code
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Use Case Example
+
+Consider the following example of a service that returns split fees for a specific collection: the fee is always set at 5%, but if the order is from a partner collection, 2% of the 5% fee is redirected to the partner's collection.
+
+```ts
+import { CalculateOrderFeeResult, parseAndVerifyRequest } from '@nft/service'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+const PLATFORM_ADDRESS = '0x...'
+const PLATFORM_FEE = 500 // equivalent to 5% with a precision of 2
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx]:
```suggestion
const PLATFORM_FEE = 500 // equivalent to 5% with a precision of 4
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+const COLLECTION_PARTNER = '0x...'
+const PARTNER_ADDRESS = '0x...'
+const PARTNER_FEE = 200 // equivalent to 2% with a precision of 2
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx]:
```suggestion
const PARTNER_FEE = 200 // equivalent to 2% with a precision of 4
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    req,
+    process.env.LITEFLOW_WEBHOOK_SECRET,
+  )
+  const fees =
+    args.collectionAddress === COLLECTION_PARTNER
+      ? [
+          { account: PLATFORM_ADDRESS, value: PLATFORM_ADDRESS - PARTNER_FEE },
+          { account: PARTNER_ADDRESS, value: PARTNER_FEE },
+        ]
+      : [{ account: PLATFORM_ADDRESS, value: PLATFORM_ADDRESS }]
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx]:
```suggestion
      : [{ account: PLATFORM_ADDRESS, value: PLATFORM_FEE }]
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  res: NextApiResponse<CalculateOrderFeeResult>,
+): Promise<void> {
+  const args = await parseAndVerifyRequest<'CALCULATE_ORDER_FEES'>(
+    req,
+    process.env.LITEFLOW_WEBHOOK_SECRET,
+  )
+  const fees =
+    args.collectionAddress === COLLECTION_PARTNER
+      ? [
+          { account: PLATFORM_ADDRESS, value: PLATFORM_ADDRESS - PARTNER_FEE },
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx]:
```suggestion
          { account: PLATFORM_ADDRESS, value: PLATFORM_FEE - PARTNER_FEE },
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx (hunk 5 above)
Comment (NicolasMahe) [reply docs/pages/service/CALCULATE_ORDER_FEES.mdx]: And it's should be converted to a string:
```suggestion
      : [{ account: PLATFORM_ADDRESS, value: PLATFORM_FEE.toString() }]
```
Would be better to use BigNumber lib

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx (hunk 6 above)
Comment (NicolasMahe) [reply docs/pages/service/CALCULATE_ORDER_FEES.mdx]:
```suggestion
          { account: PLATFORM_ADDRESS, value: (PLATFORM_FEE - PARTNER_FEE).toString() },
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+export default async function fees(
+  req: NextApiRequest,
+  res: NextApiResponse<CalculateOrderFeeResult>,
+): Promise<void> {
+  const args = await parseAndVerifyRequest<'CALCULATE_ORDER_FEES'>(
+    req,
+    process.env.LITEFLOW_WEBHOOK_SECRET,
+  )
+  const fees =
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx:59]: This example is not checking `args.feesPrecision`. If it not `4`, the returned value will be wrong.
Could we add something like:
```
PLATFORM_FEE * Math.pow(10, args.feesPrecision - 4) // adjust fee's precision to match expected precision
```

File: docs/pages/service/getStarted.mdx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+Start by determining which service you want to respond to. A list of available services can be found in the relevant sections of the documentation.
+
+## Step 2: Establish a Service Endpoint
+
+Create a publicly accessible HTTP endpoint to receive incoming requests using the POST method. This endpoint should be able to return specific payloads based on the requested service.
+
+<Tabs items={['Next.js', 'Express']}>
+<Tab>
+```ts filename="./pages/api/service.ts"
```
Comment (NicolasMahe) [inline docs/pages/service/getStarted.mdx]: It's not clear that we recommend to implement one API per type of service.
```suggestion
```ts filename="./pages/api/SERVICE_TYPE.ts"
```

File: docs/pages/service/getStarted.mdx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+}
+```
+
+</Tab>
+<Tab>
+```ts filename="./index.ts"
+import { parseAndVerifyRequest } from '@nft/service'
+import express from 'express'
+const app = express()
+app.post('/service', async (req, res) => {
```
Comment (NicolasMahe) [inline docs/pages/service/getStarted.mdx]:
```suggestion
app.post('/SERVICE_TYPE', async (req, res) => {
```

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx (hunk 7 above)
Comment (antho1404) [reply docs/pages/service/CALCULATE_ORDER_FEES.mdx:59]: That makes the code super complex to understand for something that should not be useful most of the time. What I would suggest is to keep this as it is but add a warning under with something like:
"This example uses a precision of 4, but you should always check with the argument `feesPrecision`"

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx (hunk 6 above)
Comment (antho1404) [reply docs/pages/service/CALCULATE_ORDER_FEES.mdx]: done

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx (hunk 5 above)
Comment (antho1404) [reply docs/pages/service/CALCULATE_ORDER_FEES.mdx]: done

I was wondering if it makes sense to use `BigNumber` as well but similar to my previous point, I believe this adds complexity for nothing for now

File: docs/pages/service/CALCULATE_ORDER_FEES.mdx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    args.collectionAddress === COLLECTION_PARTNER
+      ? [
+          {
+            account: PLATFORM_ADDRESS,
+            value: (PLATFORM_FEE - PARTNER_FEE).toString(),
```
Comment (NicolasMahe) [inline docs/pages/service/CALCULATE_ORDER_FEES.mdx]: I find out that `toFixed` exist on the number type. Will be nicer:
```suggestion
            value: (PLATFORM_FEE - PARTNER_FEE).toFixed(0),
```

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**